internal/blockcreator/blockcreator.go

```diff
@@ -166,12 +152,7 @@ func (b *BlockCreator) Start() {
 				b.logger.Panicf("block submission to block-replicator failed: %v", err)
 			}
 
-			h, err := blockstore.ComputeBlockBaseHash(block)
-			if err != nil {
-				b.logger.Panicf("Error calculating block hash {%v}", err)
-			}
-			b.previousBlockHeaderBaseHash = h
-			b.nextBlockNumber++
+			b.nextProposalNumber++
```

Comment:
Earlier, block creator helped to parallelise certain tasks such as computing block hash, creating the base header by reading from the block store (IO). Now, this would be done serially within the replicator for each block. If this can be addressed within the replicator, it would be nice.

Reply:
The go-routine that does that - the propose-loop go routine - is really lightly loaded, so this does not pose a performance hit in my opinion. More over, since the event-loop (which commits) and the propose-loop now communicate through memory to coordinate what was the last committed block, an access to IO is saved.

Note that the earlier version had a bug when a node was leader, lost leadership, and assumed leadership again.... the block creator never was sync'ed with the last committed block. This is the main reason behind this PR.